### PR TITLE
fix(javascript): update Predict type exports

### PR DESCRIPTION
**Pull request by francoischalifour**
## 🧭 What and Why

- 🎟 JIRA Ticket: [PRED-542](https://algolia.atlassian.net/browse/PRED-542) and [APIC-626](https://algolia.atlassian.net/browse/APIC-626)
- [Slack thread](https://algolia.slack.com/archives/C0341QDM3EG/p1660819073504399)

### Changes included:

- rename `Affinities` to `Affinity`
- export `Region` and add description
- make model types' (`Affinities`, `FunnelStage`) keys required (they shouldn't be optional)
- export `ModelsToRetrieve` as `ModelsToRetrieveEnum[]`
- export `TypesToRetrieve` as `TypesToRetrieveEnum[]`

## 🧪 Test

None added.

### CODE REVIEW

File: specs/predict/paths/fetchUserProfile.yml [hunk 1]
```diff
@@ -65,18 +65,21 @@ post:
                       value:
                         type: array
                         items:
-                          title: affinities
+                          title: affinity
                           type: object
                           properties:
```
Comment (shortcuts) [inline specs/predict/paths/fetchUserProfile.yml:70]: `required` must be set at the same level of `properties` not the property itself

File: specs/predict/common/schemas/Params.yml [hunk 2]
```diff
@@ -1,38 +1,38 @@
...
 
-modelsToRetrieve:
+modelsToRetrieveParam:
   type: object
   required:
-    - modelsToRetrieve
+    - modelsToRetrieveParam
   properties:
-    modelsToRetrieve:
+    modelsToRetrieveParam:
```
[the remark this answers is not in the document]
Comment (millotp) [reply specs/predict/common/schemas/Params.yml:17]: should we use `Params` here ?

[the remark this answers is not in the document]
Comment (shortcuts) [reply specs/predict/common/schemas/Params.yml:17]: cc @francoischalifour is the `Param` suffix correct here? Considering it's the name of the property

[the remark this answers is not in the document]
Comment (francoischalifour) [reply specs/predict/common/schemas/Params.yml:17]: Both names are fine, but `modelsToRetrieveParam` is a bit more semantically correct to me.

[the remark this answers is not in the document]
Comment (shortcuts) [reply specs/predict/common/schemas/Params.yml:17]: Yes but those properties are the one sent to the API

[the remark this answers is not in the document]
Comment (shortcuts) [reply specs/predict/common/schemas/Params.yml:17]: https://github.com/algolia/api-clients-automation/commit/fbec4ab7c53ed35d1bfe2e69afba238b87cdccb2#diff-fda81996d02c599377a2443466c81de279e1b4e8afdf279fe20db49307cdc7a6R42

**Review by shortcuts — APPROVED** [review]
LGTM!

**Pull request merged**